### ISSUE TITLE
[v2] useField would love to have deprecated setFieldValue

### ISSUE THREAD

**Issue by Nomeasmo** [jaredpalmer/formik#1907]
### Current Behavior

Currently useField returns onChange, which expects synthetic events and calls after some lengthy parsing of the event the setFieldValue method.

### Desired Behavior

In order support easier creation of custom controls as well as the use from third party libraries it would be advantageous to have a setFieldValue method again.

### Suggested Solution

I'd extend the useFields meta object with a reference to that function.

### Who does this impact? Who is this for?

People using non standard inputs, like custom fields or if third party library is not completely doing what expected.

Some examples:
```js
// Semantic UI 
<Dropdown
   {...field}
   onChange={(_, { value }) => field.onChange({ target: { id, name, value } })}
   ...
/>
// or
const onModifiedChange = (event) => {
    // Not sure either the synthetic event of semantic ui checkbox is wrong or the
    // getValueForCheckbox from Formik is not ok.
    event.target.value = event.target.checked.toString();
    field.onChange(event);
};
return (
    <Checkbox
       onChange={onModifiedChange}
// Or handcrafted
     <input
         type="button"
         onChange={() => field.onChange({ target: { id: fieldId, name, value } })}
```
### Describe alternatives you've considered

See above

### Additional context

n/a

**Comment by jaredpalmer** [comment]
Word. For now, just use `useFormikContext()`

**Comment by jaredpalmer** [comment]
We could do this:

```tsx
// Add within useFormik()
const getFieldHelpers = React.useCallback(
  (name: string) => {
    return {
      setValue: (value: any) => setFieldValue(name, value),
      setError: (value: any) => setFieldError(name, value),
      setTouched: (value: boolean) => setFieldTouched(name, value)
    };
  },
  [setFieldError, setFieldValue, setFieldTouched]
);
```

```diff
const MyInput = props => {
-  const form = useFormikContext();
-  const [field, meta] = useField(props);
+  const [field, meta, helpers] = useField(props);
  return (
    <input
      {...field}
      onChange={e => {
-        formik.setFieldValue(props.name, e.target.value);
+        helpers.setValue(e.target.value);
      }}
    />
  );
};
```

**Comment by killjoy2013** [comment]
Hi,
My project starts to complain about `setFieldValue` after 2.0.1 upgrade.

Can I simply use 
`formikProps.setValues({[e.target.name]: e.target.value})`
instead of 
`formikProps.setFieldValue(e.target.name, e.target.value);`

Thanks for this new version. I really needed hooks in Formik..

**Comment by kakadiadarpan** [comment]
> We could do this:
> 
> ```tsx
> // Add within useFormik()
> const getFieldHelpers = React.useCallback(
>   (name: string) => {
>     return {
>       setValue: (value: any) => setFieldValue(name, value),
>       setError: (value: any) => setFieldError(name, value),
>       setTouched: (value: boolean) => setFieldTouched(name, value)
>     };
>   },
>   [setFieldError, setFieldValue, setFieldTouched]
> );
> ```
> 
> ```diff
> const MyInput = props => {
> -  const form = useFormikContext();
> -  const [field, meta] = useField(props);
> +  const [field, meta, helpers] = useField(props);
>   return (
>     <input
>       {...field}
>       onChange={e => {
> -        formik.setFieldValue(props.name, e.target.value);
> +        helpers.setValue(e.target.value);
>       }}
>     />
>   );
> };
> ```

@jaredpalmer That's really neat! I would love to take this up if no one is already working on this 🙂

PS: I love Formik! Thank you for all the work you put in here. 😄

**Comment by jaredpalmer** [comment]
Released helpers in 2.1
